### PR TITLE
Fix(eos_cli_config_gen)!: update logic in monitor_sessions to not require both source and destination

### PR DESCRIPTION
**Pull request by carlbuchmann**
## Change Summary

Monitor-session templates expect both keys monitor_session.sources is arista.avd.defined and monitor_session.destinations to be defined before generating any config but this is not required in EOS, so relaxing this check and requirement.

```jinja2
{% if monitor_sessions is arista.avd.defined %}
!
{%     for monitor_session in monitor_sessions | arista.avd.natural_sort('name') if monitor_session.name is arista.avd.defined %}
{%         if monitor_session.sources is arista.avd.defined and monitor_session.destinations is arista.avd.defined %} # <- remove this if condition.
```
## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Update logic and optimize tests in template.
Update indentation.
Updated eos_designs_unit test case where monitor session is configured without destination, and now we see configuration being generated.

## How to test

All Molecule tests should pass!

### CODE REVIEW

**Review by ClausHolbechArista — APPROVED** [review]

**Comment by gmuloc** [comment]
As discussed on chat with @ClausHolbechArista, this PR could trigger some changes for existing deployment with only source pr destination that will suddenly start to generate config (hence a "breaking change").

I would be in favor of:
1/ Adding a `!` indicating it is breaking (to our output) but it is a bug fix so it is allowed by SemVer
2/ maybe pushing this to 5.0.0 or if we believe we can't wait then making it a patch with an addition in the release notes but we may be argued that we don't follow SemVer stricto sensu

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/monitor-sessions.j2 [hunk 1]
```diff
@@ -9,48 +9,46 @@
...
-{%             endfor %}
+{%         endfor %}
 
 ####### {{ monitor_session.name }} Destinations and Session Settings
 
 | Settings | Values |
 | -------- | ------ |
-| Destinations | {{ monitor_session.destinations | join(', ') }} |
-{%             if monitor_session.encapsulation_gre_metadata_tx is arista.avd.defined(true) %}
+| Destinations | {{ monitor_session.destinations | arista.avd.default("-") | join(', ') }} |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/monitor-sessions.j2]:
```suggestion
| Destinations | {{ monitor_session.destinations | arista.avd.default(["-"]) | join(', ') }} |
```

**Review by sugetha24 — APPROVED** [review]
LGTM

**Comment by gmuloc** [comment]
@carlbuchmann I think this one should be marked as breaking right? (`!` in the PR title)

**Review by ClausHolbechArista — APPROVED** [review]

**Comment by gmuloc** [comment]
@carlbuchmann we need to add something in the release notes for the change of behavior for 5.x.x

**Review by gmuloc — CHANGES_REQUESTED** [review]
blocking until release notes are updated and conflict resolved

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 2]
```diff
@@ -702,6 +702,12 @@ The following `eos_cli_config_gen` inputs can be given as needed to generate the
 All changes except `"RANCID-CONTENT-TYPE: arista"` have been incorporated into `eos_designs`. Action is only required when using `eos_cli_config_gen` directly,
 or if the `RANCID-CONTENT-TYPE: arista` is required.
 
+### `monitor_sessions` no longer requires both `monitor_sessions.sources` and `monitor_sessions.destinations` to be defined for generating configuration
+
+In AVD 4.0.0, `monitor_sessions` required both `monitor_sessions.sources` and `monitor_sessions.destinations` for configuration to be generated, but this is not required by EOS.
+
+As of avd 5.0.0 `monitor_sessions` configuration will be generated if either `monitor_sessions.sources` or `monitor_sessions.destinations` are not defined. To revert this behavior update your inventory to remove the configuration.
+
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
As of avd 5.0.0 `monitor_sessions` configuration will be generated even if `monitor_sessions.sources` or `monitor_sessions.destinations` are not defined. To revert this behavior update your inventory to remove the configuration.

```

**Review by gmuloc — APPROVED** [review]

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md [hunk 3]
```diff
@@ -80,7 +80,11 @@ As of AVD 5.0.0 the automatic conversion from both of the previous models has be
 
 See the [porting guide](../porting-guides/5.x.x.md#no-auto-conversion-of-old-data-model-for-router_bgpredistribute_routes-and-router_bgpvrfsredistribute_routes) for details.
 
-### `router_isis.address_family_ipv4/6.enabled` is now `Required` to generate the configuration and documentation for router_isis address-families
+### `monitor_sessions` no longer requires both `monitor_sessions.sources` and `monitor_sessions.destinations` to be defined for generating configuration
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/release-notes/5.x.x.md]:
```suggestion
#### `monitor_sessions` no longer requires both `monitor_sessions.sources` and `monitor_sessions.destinations` to be defined for generating configuration
```

**Pull request merged**